network-fs.go

```diff
@@ -0,0 +1,239 @@
+		if err.Error() == errVolumeNotFound.Error() {
+			return errVolumeNotFound
+		}
+		return err
+	}
+	return nil
+}
+
+// File operations.
+
+// CreateFile - create file.
+func (n networkFS) CreateFile(volume, path string) (writeCloser io.WriteCloser, err error) {
+	writeURL := new(url.URL)
+	writeURL.Scheme = "http" // TODO fix this.
+	writeURL.Host = n.netAddr
+	writeURL.Path = fmt.Sprintf("/minio/rpc/storage/upload/%s", urlpath.Join(volume, path))
+
+	contentType := "application/octet-stream"
+	readCloser, writeCloser := io.Pipe()
+	go func() {
+		resp, err := n.httpClient.Post(writeURL.String(), contentType, readCloser)
+		if err != nil {
+			readCloser.CloseWithError(err)
+			return
+		}
+		if resp != nil {
+			if resp.StatusCode != http.StatusNotFound {
+				readCloser.CloseWithError(errFileNotFound)
+				return
+			}
+			readCloser.CloseWithError(errors.New("Invalid response."))
+		}
+	}()
+	return writeCloser, nil
+}
+
+// StatFile - get latest Stat information for a file at path.
+func (n networkFS) StatFile(volume, path string) (fileInfo FileInfo, err error) {
+	if err = n.rpcClient.Call("Storage.StatFileHandler", StatFileArgs{
+		Vol:  volume,
+		Path: path,
+	}, &fileInfo); err != nil {
+		if err.Error() == errVolumeNotFound.Error() {
+			return FileInfo{}, errVolumeNotFound
+		} else if err.Error() == errFileNotFound.Error() {
+			return FileInfo{}, errFileNotFound
+		} else if err.Error() == errIsNotRegular.Error() {
+			return FileInfo{}, errFileNotFound
+		}
+		return FileInfo{}, err
+	}
+	return fileInfo, nil
+}
+
+// ReadFile - reads a file.
+func (n networkFS) ReadFile(volume string, path string, offset int64) (reader io.ReadCloser, err error) {
+	readURL := new(url.URL)
+	readURL.Scheme = "http" // TODO fix this.
+	readURL.Host = n.netAddr
+	readURL.Path = fmt.Sprintf("/minio/rpc/storage/download/%s", urlpath.Join(volume, path))
+	readQuery := make(url.Values)
+	readQuery.Set("offset", strconv.FormatInt(offset, 10))
+	readURL.RawQuery = readQuery.Encode()
+	resp, err := n.httpClient.Get(readURL.String())
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusNotFound {
+			return nil, errFileNotFound
+		}
+		return nil, errors.New("Invalid response")
+	}
+	return resp.Body, nil
+}
+
+// ListFiles - List all files in a volume.
+func (n networkFS) ListFiles(volume, prefix, marker string, recursive bool, count int) (files []FileInfo, eof bool, err error) {
+	listFilesReply := ListFilesReply{}
+	if err = n.rpcClient.Call("Storage.ListFilesHandler", ListFilesArgs{
+		Vol:       volume,
+		Prefix:    prefix,
+		Marker:    marker,
+		Recursive: recursive,
+		Count:     count,
+	}, &listFilesReply); err != nil {
+		if err.Error() == errVolumeNotFound.Error() {
+			return nil, true, errVolumeNotFound
+		}
+		return nil, true, err
+	}
+	// List of files.
+	files = listFilesReply.Files
+	// EOF.
+	eof = listFilesReply.EOF
+	return files, eof, nil
```

Comment:
you could just do `return` or `return listFilesReply.Files, listFilesReply.EOF, nil`. Current way is confusing.